pandas/tests/io/json/test_pandas.py

```diff
@@ -82,76 +80,79 @@ def setup(self, datapath):
         del self.tsframe
         del self.mixed_frame
 
-    def test_frame_double_encoded_labels(self):
+    def test_frame_double_encoded_labels(self, orient):
         df = DataFrame(
             [["a", "b"], ["c", "d"]],
             index=['index " 1', "index / 2"],
             columns=["a \\ b", "y / z"],
         )
 
-        assert_frame_equal(df, read_json(df.to_json(orient="split"), orient="split"))
-        assert_frame_equal(
-            df, read_json(df.to_json(orient="columns"), orient="columns")
-        )
-        assert_frame_equal(df, read_json(df.to_json(orient="index"), orient="index"))
-        df_unser = read_json(df.to_json(orient="records"), orient="records")
-        assert_index_equal(df.columns, df_unser.columns)
-        tm.assert_numpy_array_equal(df.values, df_unser.values)
+        result = read_json(df.to_json(orient=orient), orient=orient)
+        expected = df.copy()
 
-    def test_frame_non_unique_index(self):
-        df = DataFrame([["a", "b"], ["c", "d"]], index=[1, 1], columns=["x", "y"])
+        if orient == "records" or orient == "values":
+            expected = expected.reset_index(drop=True)
+        if orient == "values":
+            expected.columns = range(len(expected.columns))
 
-        msg = "DataFrame index must be unique for orient='index'"
-        with pytest.raises(ValueError, match=msg):
-            df.to_json(orient="index")
-        msg = "DataFrame index must be unique for orient='columns'"
-        with pytest.raises(ValueError, match=msg):
-            df.to_json(orient="columns")
+        assert_frame_equal(result, expected)
 
-        assert_frame_equal(df, read_json(df.to_json(orient="split"), orient="split"))
-        unser = read_json(df.to_json(orient="records"), orient="records")
-        tm.assert_index_equal(df.columns, unser.columns)
-        tm.assert_almost_equal(df.values, unser.values)
-        unser = read_json(df.to_json(orient="values"), orient="values")
-        tm.assert_numpy_array_equal(df.values, unser.values)
+    @pytest.mark.parametrize(
+        "orient,expected",
+        [
+            (
+                "split",
+                DataFrame([["a", "b"], ["c", "d"]], index=[1, 1], columns=["x", "y"]),
+            ),
+            ("records", DataFrame([["a", "b"], ["c", "d"]], columns=["x", "y"])),
+            ("values", DataFrame([["a", "b"], ["c", "d"]])),
```

Comment:
would these be clearer if they conditionally did reset_index/columns in the test method to match the behavior in other tests?  fine by me either way

Reply:
No yea this is also a great idea - would match the pattern in other tests better for sure